src/main/java/com/toamistaketracker/ToaMistake.java

```diff
@@ -125,39 +135,42 @@ public static ToaMistake toGroupedMistake(ToaMistake mistake) {
+     * settings about whether to stack question marks and the current mistake count of either
+     * this specific mistake in the raid for the current raider, *or* the total current raid mistake
+     * count for the current raider.
      *
      * @param mistake      The mistake
-     * @param mistakeCount The current mistake count of this mistake for the current raider in this current raid,
-     *                     previously before this.
+     * @param mistakeCount The current mistake count of this mistake *or* the current raid mistake count for the
+     *                     raider in this raid
+     * 
      * @return The mistake chat message to use for the raider
      */
-    public static String getChatMessageForMistakeCount(ToaMistake mistake, int mistakeCount) {
-        // Special case a few mistake chat messages
-        if (mistake == AKKHA_UNSTABLE_ORB) {
-            return getChatMessageForAkkhaUnstableOrb(mistake.getChatMessage(), mistakeCount);
-        } else if (mistake == BABA_SLAM) {
-            return getChatMessageForBabaSlam(mistake.getChatMessage(), mistakeCount);
+    public static String getChatMessageForMistakeCount(ToaMistakeTrackerConfig config, ToaMistake mistake,
+            int mistakeCount) {
+        String mistakeMessage = mistake.getChatMessage(config);
+        // Special case a few mistake chat messages based on config
+        if (config.mistakeMessageStacking() != StackingBehavior.NONE && mistakeMessage.equals("?")) {
+            return getStackingChatMessage(mistakeMessage, mistakeCount);
+        } else if (mistakeMessage.contains("|")) {
+            return getAlternatingChatMessage(mistakeMessage, mistakeCount);
         }
 
-        return mistake.getChatMessage();
+        return mistakeMessage;
     }
 
-    private static String getChatMessageForAkkhaUnstableOrb(String message, int mistakeCount) {
+    private static String getStackingChatMessage(String message, int mistakeCount) {
         // Keep adding ?s to the text for every subsequent mistake in this raid. A bit hacky but it's funny.
         StringBuilder sb = new StringBuilder(message);
-        for (int i = 0; i < Math.min(mistakeCount, MAX_AKKHA_ORBS_CHAT_MESSAGE_LENGTH); i++) {
+        for (int i = 0; i < Math.min(mistakeCount, MAX_STACKING_CHAT_MESSAGE_LENGTH); i++) {
             sb.append(message);
         }
         return sb.toString();
     }
 
-    private static String getChatMessageForBabaSlam(String message, int mistakeCount) {
-        if (mistakeCount % 2 == 0) {
-            return message;
-        } else {
-            return ALTERNATE_BABA_SLAM_CHAT_MESSAGE;
-        }
+    private static String getAlternatingChatMessage(String message, int mistakeCount) {
+        String[] messageChoices = message.split("\\|");
+        String messageChoice = messageChoices[mistakeCount % messageChoices.length];
+        return messageChoice;
```

Comment:
Change funnels the user-inputted message through a stream which splits the input by "|", then filters all empty messages to prevent things like "|||f|||" from looking like: `["", "", "f", "",  ""]` and instead becoming a single choice of `["f"]`. This still allows things like "f| " which would become: `["f", " "]` since a space message is not empty.

Additionally, if the input message splits into an empty array (aka the input is just "|" or "|||||" etc) then the default message is shown instead.